pilot/cmd/pilot-agent/status/server.go

```diff
@@ -527,6 +528,23 @@ func (s *Server) handleQuit(w http.ResponseWriter, r *http.Request) {
 	notifyExit()
 }
 
+func (s *Server) handleTcpProbe(w http.ResponseWriter, req *http.Request) {
+	portStr := strings.TrimPrefix(req.URL.Path, "/tcp-probe/")
+
+	port, err := strconv.Atoi(portStr)
```

Comment:
This is not secure, as it allows connecting to arbitrary ports. We should follow the model in handleAppProbe that has a strict allow-list of legal requests